bundles/org.openhab.binding.nikobus/src/main/java/org/openhab/binding/nikobus/internal/handler/NikobusPcLinkHandler.java

```diff
@@ -0,0 +1,330 @@
+            return;
+        }
+
+        NikobusCommand.ResponseHandler responseHandler = command.getResponseHandler();
+        if (responseHandler == null) {
+            logger.debug("No response expected for current command");
+            return;
+        }
+
+        if (ack == null) {
+            logger.debug("No ack received");
+            return;
+        }
+
+        String requestCommandId = command.getPayload().substring(3, 5);
+        String ackCommandId = ack.substring(3, 5);
+        if (!ackCommandId.equals(requestCommandId)) {
+            logger.debug("Unexpected command's ack '{}' != '{}'", requestCommandId, ackCommandId);
+            return;
+        }
+
+        // Check if response has expected length.
+        if (commandPayload.length() != responseHandler.getResponseLength()) {
+            logger.debug("Unexpected response length");
+            return;
+        }
+
+        if (!commandPayload.startsWith(responseHandler.getResponseCode())) {
+            logger.debug("Unexpected response command code");
+            return;
+        }
+
+        String requestCommandAddress = command.getPayload().substring(5, 9);
+        String ackCommandAddress = commandPayload.substring(responseHandler.getAddressStart(),
+                responseHandler.getAddressStart() + 4);
+        if (!requestCommandAddress.equals(ackCommandAddress)) {
+            logger.debug("Unexpected response address");
+            return;
+        }
+
+        if (responseHandler.complete(commandPayload)) {
+            resetProcessingAndProcessNext();
+        }
+    }
+
+    public void sendCommand(NikobusCommand command) {
+        synchronized (pendingCommands) {
+            pendingCommands.addLast(command);
+        }
+
+        scheduler.submit(this::processCommand);
+    }
+
+    @SuppressWarnings({ "unused", "null" })
+    private void processCommand() {
+        NikobusCommand command;
+        synchronized (pendingCommands) {
+            if (currentCommand != null) {
+                return;
+            }
+
+            command = pendingCommands.pollFirst();
+            if (command == null) {
+                return;
+            }
+
+            currentCommand = command;
+        }
+        sendCommand(command, 3);
+    }
+
+    private void sendCommand(NikobusCommand command, int retry) {
+        logger.debug("Sending retry = {}, command '{}'", retry, command.getPayload());
+
+        NikobusConnection connection = this.connection;
+        if (connection == null) {
+            return;
+        }
+
+        try {
+            connectIfNeeded(connection);
+
+            OutputStream outputStream = connection.getOutputStream();
+            if (outputStream == null) {
+                return;
+            }
+            outputStream.write(command.getPayload().getBytes());
+            outputStream.flush();
+        } catch (IOException e) {
+            logger.error("Sending command failed due {}", e.getMessage(), e);
```

Comment:
Can be debug as the status is also set.
```suggestion
            logger.debug("Sending command failed due {}", e.getMessage(), e);
```